packages/stream-model/src/model.ts

```diff
@@ -72,10 +72,31 @@ async function throwReadOnlyError(): Promise<void> {
  * there can be only one instance of this model per account (if a new instance is created it
  * overrides the old one).
  */
-export enum ModelAccountRelation {
-  LIST = 'list',
-  SINGLE = 'single',
-}
+export type ModelAccountRelation = { type: 'list' } | { type: 'single' }
+
+/**
+ * Identifies types of properties that are supported as relations by the indexing service.
+ *
+ * Currently supported types of relation properties:
+ * - 'account': references a DID property
+ * - 'document': references a SteamID property with associated 'model' the related document must use
+ *
+ */
+export type ModelRelationDefinition = { type: 'account' } | { type: 'document'; model: string }
+
+/**
+ * A mapping between model's property names and types of relation properties
+ *
+ * It indicates which properties of a model are relation properties and of what type
+ */
+export type ModelRelationsDefinition = Record<string, ModelRelationDefinition>
+
+export type ModelDocumentViewDefinition = { type: 'documentAccount' } | { type: 'documentVersion' }
```

Comment:
`ModelDocumentMetadataViewDefinition`?  Too verbose?

Reply:
@stbrody what do you think about future support for a "document history" view that represents a collection of all the versions of the document please?
I don't think it would really fit the "metadata" naming convention, do you? Should we use `ModelDocumentMetadataViewDefinition` and `ModelDocumentViewDefinition = ModelDocumentMetadataViewDefinition | { type: 'documentHistory' }` for example?

Reply:
yeah, that would seem sufficiently different to me that it merits its own category.